### PR TITLE
autcomplete remote branches

### LINKED ISSUE #22: autocomplete remote branches
In this WIP

https://github.com/albfan/clink-completions/commit/d48adfbffed3dc5909a5a93d20e4ee0622efa8a2

I found that clink divides autocompletion on "/".

    $ git rebase <TAB>
    origin/master origin/wip vladimir/master vladimir/extended vladimir/dev master wip
    $ git rebase o<TAB>
    $ git rebase origin/<TAB>
    -- no output --

and it do not offer autocompletion

When there is only one branch on remote autocompletion works.

Is there any solution to avoid stop on "/"? I've tried "disable_file_matching()" but didn't solve it.

### PR DESCRIPTION
**Pull request by albfan**
resolves #22

### CODE REVIEW

File: git.lua [hunk 1]
```diff
@@ -142,6 +203,11 @@ local function checkout_spec_generator(token)
         table.insert(res_filter, '*' .. branch)
     end
 
+    for _,branch in ipairs(local_or_remote_branches(token)) do
```
Comment (vladimir-kotikov) [inline git.lua]: By just adding this block you're duplicating local branches in completions table. Consider removal the block above (that responsible for local branches only)

File: git.lua [hunk 2]
```diff
@@ -594,35 +661,35 @@ local git_parser = parser(
         "submodule",
         "subtree",
         "svn"..parser({
-		    "init"..parser("-T", "--trunk", "-t", "--tags", "-b", "--branches", "-s", "--stdlayout",
```
Comment (vladimir-kotikov) [inline git.lua]: Please remove this change. The whitespace changes is not a part of feature, you adding :)

File: git.lua [hunk 3]
```diff
@@ -112,6 +112,30 @@ local function branches(token)
     return res
 end
 
+local function alias(token)
```
Comment (vladimir-kotikov) [inline git.lua]: Please submit 'alias' feature as a separate pull request. Also i'm not verya familiar with this, so could you please clarify the purpose of this?

**Comment by vladimir-kotikov** [comment]
Thanks for your contribution again, @albfan. Could you please update the PR and squash the commits into one before merging.

File: git.lua (hunk 3 above)
Comment (albfan) [inline git.lua]: See this

http://stackoverflow.com/a/22875140/848072

or my dotfiles

https://github.com/albfan/dotfiles/blob/master/gitconfig#L3

or related doc

https://git.wiki.kernel.org/index.php/Aliases#Aliases

alias are contractions to ease execution of regular basis commands

**Comment by albfan** [comment]
Why squash them? They 're totally unrelated.

I will remove unrelated to this PR and wait for you to merge it to propose others

**Comment by albfan** [comment]
Now you can merge only this feature

**Pull request closed without merging**